src/Neo/SmartContract/Native/NamedCurveHash.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2015-2024 The Neo Project.
+//
+// NamedCurveHash.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+namespace Neo.SmartContract.Native
+{
+    /// <summary>
+    /// Represents a pair of the named curve used in ECDSA and a hash algorithm used to hash message.
+    /// This is a compatible extension of an obsolete <see cref="NamedCurve"/> enum.
+    /// </summary>
+    /// <remarks>
+    /// https://tools.ietf.org/html/rfc4492#section-5.1.1
+    /// </remarks>
+    public enum NamedCurveHash : byte
```

Comment:
Maybe two different enums is better than this pair, no?

Reply:
We need native CryptoLib’s werifyWithECDsa method to be compatible with the old one, since there are transactions that call this method. and thus, we use a pair here.

Reply:
Theoretically, `VerifyWithECDsa` could be changed to have two parameters in some HF. But practically, #3210.